### PR TITLE
Utilize Hide Hidden Entities

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

CleanShot 2022-03-14 at 14 25 10

CleanShot 2022-03-14 at 14 24 57

CleanShot 2022-03-14 at 14 24 34

Not used in Generated Lovelace

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/devices/device-detail/ha-device-entities-card.ts [hunk 1]
```diff
@@ -60,77 +60,75 @@ export class HaDeviceEntitiesCard extends LitElement {
   }
 
   protected render(): TemplateResult {
-    const disabledEntities: EntityRegistryStateEntry[] = [];
+    if (!this.entities.length) {
+      return html`
+        <div class="empty card-content">
```
Comment (balloob) [inline src/panels/config/devices/device-detail/ha-device-entities-card.ts:65]: You're not rendering `<ha-card>` but we previously did.

**Comment by emontnemery** [comment]
Hiding or disabling entities is not a feature integrations need to implement, I don't think we should have these warnings?
image

**Comment by frenck** [comment]
image

Should that say: "hidden and disabled entities"?

**Comment by zsarnett** [comment]
> image
> 
> Should that say: "hidden/disabled entities"?

I don't think so as we use hidden entities on the entities table page as well. I thought about this a lot and this way just means we can add other things as well. They are hidden from the user being disabled or hidden.

**Comment by zsarnett** [comment]
CleanShot 2022-03-14 at 20 37 00

File: src/panels/config/devices/device-detail/ha-device-entities-card.ts [hunk 2]
```diff
@@ -60,77 +60,77 @@ export class HaDeviceEntitiesCard extends LitElement {
...
+                    hiddenEntities.length
+                  )}
+                </button>
+              `
+            : html`
+                ${hiddenEntities.map((entry) => this._renderEntry(entry))}
+                <button class="show-more" @click=${this._toggleShowHidden}>
                   ${this.hass.localize(
-                    "ui.panel.config.devices.entities.add_entities_lovelace"
+                    "ui.panel.config.devices.entities.hide_disabled"
```
Comment (balloob) [inline src/panels/config/devices/device-detail/ha-device-entities-card.ts:115]: it's no longer just hiding disabled

File: src/panels/config/entities/entity-registry-settings.ts [hunk 3]
```diff
@@ -211,75 +215,97 @@ export class EntityRegistrySettings extends SubscribeMixin(LitElement) {
...
+          .header=${this.hass.localize(
+            "ui.dialogs.entity_registry.editor.advanced"
+          )}
+          outlined
+        >
+          <div class="label">
+            ${this.hass.localize(
+              "ui.dialogs.entity_registry.editor.view_status"
+            )}:
+          </div>
```
Comment (balloob) [inline src/panels/config/entities/entity-registry-settings.ts:228]: If disabled by integration, let's include a sentence why the user can't pick anything here.

File: src/translations/en.json [hunk 4]
```diff
@@ -102,6 +102,12 @@
       "integration": "Integration",
       "config_entry": "Config entry",
       "device": "Device"
+    },
+    "hidden_by": {
+      "user": "User",
+      "integration": "Integration",
+      "config_entry": "Config entry",
```
Comment (balloob) [inline src/translations/en.json:109]: I don't think that we have this one (yet)

File: src/panels/config/entities/ha-config-entities.ts [hunk 5]
```diff
@@ -101,6 +101,8 @@ export class HaConfigEntities extends SubscribeMixin(LitElement) {
```
Comment (balloob) [inline src/panels/config/entities/ha-config-entities.ts:101]: You will need to show a `hidden` icon in the `status` column.

File: src/translations/en.json [hunk 6]
```diff
@@ -783,13 +789,18 @@
...
+          "disabled_label": "Disabled",
           "enabled_cause": "Disabled by {cause}.",
+          "hidden_label": "Hidden",
+          "hidden_cause": "Hidden by {cause}.",
           "device_disabled": "The device of this entity is disabled.",
           "open_device_settings": "Open device settings",
           "enabled_description": "Disabled entities will not be added to Home Assistant.",
           "enabled_delay_confirm": "The enabled entities will be added to Home Assistant in {delay} seconds",
           "enabled_restart_confirm": "Restart Home Assistant to finish enabling the entities",
+          "hidden_description": "Hidden entities will not be shown in Home Assistant UI.",
```
Comment (balloob) [inline src/translations/en.json:803]:
```suggestion
          "hidden_description": "Hidden entities will not be shown. Their history is still tracked and they can still be targeted with services.",
```

Comment (matthiasdebaat) [reply src/translations/en.json:803]: How about `Hidden entities will not be shown on your dashboard. Their history is still tracked and you can still interact with them with services.`

Comment (zsarnett) [reply src/translations/en.json:803]: I'm no longer using this but I guess we should add it back

Comment (balloob) [reply src/translations/en.json:803]: Can we use it here?

image

File: src/panels/config/entities/ha-config-entities.ts [hunk 7]
```diff
@@ -280,6 +288,10 @@ export class HaConfigEntities extends SubscribeMixin(LitElement) {
                       ? this.hass.localize(
                           "ui.panel.config.entities.picker.status.disabled"
                         )
+                      : entity.hidden_by
+                      ? this.hass.localize(
+                          "ui.panel.config.entities.picker.status.hidden"
```
Comment (balloob) [inline src/panels/config/entities/ha-config-entities.ts:293]: This one doesn't work

image

Comment (zsarnett) [reply src/panels/config/entities/ha-config-entities.ts:293]: oh whoops

File: src/translations/en.json (hunk 6 above)
Comment (zsarnett) [reply src/translations/en.json:803]: Thats the disabled description not hidden

**Review by balloob — APPROVED** [review]

**Pull request merged**